### ISSUE TITLE
Server.properties and whitelist.json reset every time server restarts

### ISSUE THREAD

**Issue by platcham** [crowbarmaster/BedrockManagementService#50]
Hello! First, thanks for the amazing utility! I am on the latest build, 2.6.

I'm having an issue where my server.properties and whitelist.json are resetting to the default every time the server restarts. The whitelist actually resets to default when I look at the player manager in the client as well, even though the player manager has several of the players that I want on my server. 

I've tried editing the server properties through the client as well as directly with a text editor, both are still overwritten by the default on every start.

I was able to partially mitigate this issue by setting the stock_props.conf in the servers folder to the settings for my server, but that isn't ideal if I want to run multiple servers. 

I tried fixing the whitelist by setting the file to read only, but that just results in the service failing to start. For now my temporary fix is keeping a copy of my full whitelist, and replacing and reloading the whitelist every time my server starts, which is tedious.

Thank you for the help!

**Comment by tiamat2012** [comment]
Have you tried modifying the files in the client that came with this service?  I think the goal is to never modify the minecraft files directly, and use the client to edit them, and it will recreate them each time the service runs (to ensure the settings in the service are always used).

**Comment by platcham** [comment]
Yes, I have tried editing both through the client and directly. Either way, the server.properties and whitelist are reset. Weirdly, the whitelist is reset to only have me on the server even though the service client player manager lists multiple people on my server correctly.

**Comment by crowbarmaster** [comment]
Hello and welcome! @tiamat2012 is correct here, `server.properties`, `whitelist.json` and `permisions.json` are all built on server start and replaced by the service. Let me try to get a couple more details first. Did these players connect first, or did you add them via the client's add new player feature? Can I see a screenshot of the player manager before you add them? You can blur or shade over the xuid/usernames if you need to. Or I welcome you to hop into discord and we can chat this out: https://discord.gg/RJ3Rn66x

**Comment by tiamat2012** [comment]
Actually, I was still on a previous beta, so I updated to 2.6 just now (It was on my todo list for today anyway).  And it reset back to the defaults for me too on reboot of the server (I rebooted to increase memory size of my VM).  So something broke between 2.56.8040.34717 and 2.6, or something else happened.  I may go back to that beta version for a short time.

**Comment by platcham** [comment]
Ah thank you, I found my issue for the whitelist. I just realized that while they were in the player manager, they were set to false for whitelisted. That's my bad. Setting whitelisted to true ensures they are actually in the whitelist when it is rebuilt.

A possible enhancement is allowing the new player registration to add a new player without the xuid. To get a new player in the server, I currently either have to disable whitelisting, have them join, then reenable whitelisting, or manually add them to the temporary whitelist so they can join, and change their permission in the player manager. Unless I'm misunderstanding that too. :P

I still have the issue of my server.properties not saving the changes I make through the client (it rebuilds from default every time), but as I'm currently only running a single server, I just set the default to what my server needs.

Thanks for helping me spot my error!

**Comment by tiamat2012** [comment]
This may be part of the issue.  In the Server/Config folder, my world .conf file is messed up:

<img width="355" alt="image" src="https://user-images.githubusercontent.com/16579793/149823550-516d694b-da0d-4955-a671-748f8ed29d6a.png">

That's all that is in there.  It's missing the server settings, which explains what @platcham sees.

@platcham Thanks for that link.  I was about to go searching for that exact same thing, as my daughter moved to college this weekend and I need to make my server Internet accessible, and find a way to let her roommates play on it safely.

**Comment by tiamat2012** [comment]
@platcham if you want to get up and running quickly for the moment - try using Beta 2.56.8040.34717.  I reverted back to that and I am back up and running (I had all my server settings saved)

**Comment by tiamat2012** [comment]
Hmm - I don't know where the binaries are for that build anymore in github.  I still had a copy of the downloaded ZIP file.  Perhaps @crowbarmaster can make it available?

**Comment by crowbarmaster** [comment]
Sure can! https://drive.google.com/file/d/1RPLaHYOPtBJ1ZbcNES4YjvEc9aLCcR-x/view?usp=sharing

I will take a look and see if I can point out the issue going on here. Stay tuned!

**Comment by crowbarmaster** [comment]
@platcham I would like to follow up on this if I could. Would you happen to have the logs still from the first start of 2.6? If you still have them, would you mind zipping that up and send it, please? The props missing in your server's conf file suggests that stock_props.conf wasn't built the first run like it should have been.

**Comment by platcham** [comment]
I do, they are attached. Note that technically these are just the first logs of when it successfully ran. I tried to run it a couple of times before I found where to accept the license. Not sure if that is relevant
[BMS First Start Logs.zip](https://github.com/crowbarmaster/BedrockManagementService/files/7898551/BMS.First.Start.Logs.zip)
.

**Comment by tiamat2012** [comment]
Understood.  Since 2.6 is somewhat unusable in the current state, perhaps release the beta version I am using as a 2.6.1 release so any new users can get up and running while you fix the bug?  You can just mention the known bugs/enhancements you already fixed in 2.6.0 in the release notes.  Or just pull 2.6 from gitlab and put up the binaries in GitHub (people don’t like having to login to google to access google drive)?

**Comment by crowbarmaster** [comment]
I got home early today and had some free time. I would like to say this was some sort of challenge but alas it seems I simply forgot to add a few lines to a method. Once things settle down I can get back to writing some more tests and keep stupid slips like this from being a pain 😄 I have a fix, tested it correctly writes the conf file, and updates from client correctly! I pushed a new public release based on this fix, as well as a couple other helpful fixes!
